src/portfft/dispatcher/subgroup_dispatcher.hpp

```diff
@@ -242,29 +235,24 @@ PORTFFT_INLINE void subgroup_impl(const T* input, T* output, const T* input_imag
           }
           global_data.log_dump_private("data loaded in registers:", priv, n_reals_per_wi);
         }
+        if (conjugate_on_load == detail::complex_conjugate::APPLIED) {
+          conjugate_inplace(priv, factor_wi);
+        }
         if (multiply_on_load == detail::elementwise_multiply::APPLIED) {
-          // Note: if using load modifier, this data need to be stored in the transposed fashion per batch to ensure
-          // low latency reads from shared memory, as this will result in much lesser bank conflicts.
-          // Tensor shape for load modifier in local memory = num_batches_in_local_mem x  FactorWI x FactorSG
-          // TODO: change the above mentioned layout to the following tenshor shape: num_batches_in_local_mem x
-          // n_ffts_in_sg x FactorWI x FactorSG
-          global_data.log_message_global(__func__, "multiplying load modifier data");
+          using vec_t = sycl::vec<T, 2>;
+          vec_t modifier_vec;
           if (working_inner) {
             PORTFFT_UNROLL
             for (Idx j = 0; j < factor_wi; j++) {
-              Idx base_offset = sub_batch * n_reals_per_fft + 2 * j * factor_sg + 2 * id_of_wi_in_fft;
-              multiply_complex(priv[2 * j], priv[2 * j + 1], loc_load_modifier_view[base_offset],
-                               loc_load_modifier_view[base_offset + 1], priv[2 * j], priv[2 * j + 1]);
+              Idx base_offset = i * n_reals_per_fft + static_cast<IdxGlobal>(sub_batch * n_reals_per_fft +
+                                                                             2 * j * factor_sg + 2 * id_of_wi_in_fft);
+              modifier_vec = *reinterpret_cast<const vec_t*>(load_modifier_data[base_offset]);
```

Comment:
Do we know that this is correctly aligned to be cast to `vec_t`?

Reply:
Also I think this might be reinterpreting the value of `load_modifier_data[base_offset]` to a pointer, is that the intention?

Reply:
Its declared as `const T* load_modifier_data = nullptr` where `T` is a scalar. I'm finding this confusing!

Reply:
its essentially reinterpreting it so that we can read two elements at once (real and imaginary part of the twiddle). Its always guaranteed to be aligned